cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -244,6 +265,54 @@ public void deleteAllApps(HttpRequest request, HttpResponder responder,
     responder.sendString(status.getCode(), status.getMessage());
   }
 
+  private BodyConsumer deployAppFromArtifact(final Id.Namespace namespace, final String appId) throws IOException {
+
+    return new AbstractBodyConsumer(File.createTempFile(appId, ".json", tmpDir)) {
```

Comment:
Do we need BodyConsumer for json file that contains artifact info and configuration?